src/EventListener/ErrorHandlerListener.php

```diff
@@ -18,16 +23,34 @@
     /** @var bool */
     private $debug;
 
-    public function __construct(ErrorHandler $errorHandler, $throwException = false, $debug = false)
-    {
+    /** @var LoggerInterface */
+    private $logger;
+
+    /** @var string */
+    private $errorLevel;
+
+    public function __construct(
+        ErrorHandler $errorHandler,
+        LoggerInterface $logger,
+        bool $throwException = false,
+        bool $debug = false,
+        string $errorLevel = LogLevel::DEBUG
+    ) {
         $this->errorHandler = $errorHandler;
         $this->throwException = $throwException;
         $this->debug = $debug;
+        $this->logger = null === $logger ? new NullLogger() : $logger;
+        $this->errorLevel = $errorLevel;
     }
 
     public function onPostExecutor(ExecutorResultEvent $executorResultEvent): void
     {
         $result = $executorResultEvent->getResult();
         $this->errorHandler->handleErrors($result, $this->throwException, $this->debug);
+        $result = $result->toArray();
+
+        if (isset($result['errors'])) {
+            $this->logger->{$this->errorLevel}(__METHOD__.' : '.\json_encode($result['errors']));
```

Comment:
You should use context instead of json_encode
```suggestion
            $this->logger->{$this->errorLevel}('GraphQL request resulted with an error.', $result);
```